datadog_lambda/tracing.py

```diff
@@ -101,14 +134,18 @@ def extract_dd_trace_context(event):
     the correct context.
     """
     global dd_trace_context
-    headers = event.get("headers", {})
-    lowercase_headers = {k.lower(): v for k, v in headers.items()}
 
-    trace_id = lowercase_headers.get(TraceHeader.TRACE_ID)
-    parent_id = lowercase_headers.get(TraceHeader.PARENT_ID)
-    sampling_priority = lowercase_headers.get(TraceHeader.SAMPLING_PRIORITY)
+    if "headers" in event:
+        trace_id, parent_id, sampling_priority = extract_context_from_http_event(event)
```

Comment:
cc: @DylanLovesCoffee I recall you are touching this function as well, and I actually suggested code changes like what @Czechh does here.

Reply:
Yup, I am! This [comment](https://github.com/DataDog/datadog-lambda-python/pull/101#discussion_r539479772)? I had changed `extract_dd_trace_context` to return the source as well and create the subsegment in a separate function.